docs/README.md

```diff
@@ -0,0 +1,358 @@
+a new object of type `EventFrame` for event collections events and `Frame` for trace
+wide events. Then the event has to be added to `TypeList` at the end of the file.
+
+### Adding new in-memory type
+
+#### Python types
+
+To load data from on-disk representation and render it in the GUI new
+in-memory type has to be added. This can be done by creating a new `InMemoryCompoundType`
+object and adding it to the `TypeList` at the end of an appropriate `in-memory.py` file.
+
+After in-memory type is created the link to the on-disk representation
+has to be created. It is done by adding following lines to the appropriate `on_disk.py` file.
+
+For event frames the syntax is:
+
+```python
+tags.dsk.tomem.add_per_event_collection_tags(
+  on_disk_type,
+  aftermath.types.x.y.z.in_memory.in_memory_type,
+  "collection_id")
+```
+
+And for trace frames:
+
+```ptyhon
+tags.dsk.tomem.add_per_trace_array_tags(
+  on_disk_type,
+  aftermath.type.x.y.z.in_memory.in_memory_type)
+```
+
+#### Source files
+
+Next some .c and .h files has to be updated so appropriate classes and functions
+are generated for the new type. It can be achieved with following steps:
+
+1. Add an array definition for the in-memory type by creating a new file
+called `libaftermath-core/src/<type>_array.h`:
+
+    ```c
+    #ifndef AM_<TYPE>_ARRAY_H
+    #define AM_<TYPE>_ARRAY_H
+
+    #include <aftermath/core/typed_array.h>
+    #include <aftermath/core/in_memory.h>
+
+    AM_DECL_TYPED_ARRAY_WITH_ELEMENT_DESTRUCTOR(
+      am_<type>_array,
+      struct am_<type>,
+      am_<type>_destroy)
+
+    #endif
+    ```
+
+    If the type has no destructor then `AM_DECL_TYPED_ARRAY(am_<type>_array, struct am_<type>)`
+    is sufficient.
+
+2. Include the new header file in
+`libaftermath-core/src/defs/aftermath/templates/toplevel/on_disk.tpl.c `.
+
+3. Add the array definition to the default array registry as follow in
+`libaftermath-core/src/default_trace_array_registry.c`:
+
+    ```c
+    #include <aftermath/core/<type>_array.h>
+
+    ...
+
+    AM_DECL_DEFAULT_ARRAY_REGISTRY_FUNCTIONS(am_<type>_array)
+    ```
+
+    And then in the same file add following in `am_build_default_trace_array_registry()`:
+
+    ```c
+    AM_DEFAULT_ARRAY_REGISTRY_REGISTER(r, am_<type>_array,
+                                       "am::<model>::<type>")
+    ```
+
+4. Add a symbolic link from `libaftermath-core/headers/aftermath/core/<type>_array.h` to
+`libaftermath-core/src/<type>_array.h`:
+
+    ```
+    ln -s --relative libaftermath-core/src/<group>_<type>_array.h libaftermath-core/headers/aftermath/core/<group>_<type>_array.h
+    ```
+
+5. Add `aftermath/core/<type>_array.h` to `nobase_include_HEADERS` in `libaftermath-core/headers/Makefile.am`
+
+6. Add `src/<type>_array.h` to `libaftermath_core_la_SOURCES` in `libaftermath-core/Makefile.am`
+
+# Using libaftermath-render
+
+## Add new type to the render
+
+To enable rendering of newly defined type in Aftermath GUI it has to be added to the
```

Comment:
Suggestion: To enable rendering of a newly-defined type in the Aftermath GUI, a new renderer needs to be added to libaftermath-render. This can be done as follows: